Change 'autoload' option values in various files

pbearne opened the pull request:
Updated values of 'autoload' option from 'yes' or 'no' to 'on' or 'off' respectively, across several PHP files. This change was implemented in the functions handling options and lock settings, enhancing clarity and consistency of the code.

Trac ticket: https://core.trac.wordpress.org/ticket/61045

Commit:
Change 'autoload' option values in various files

Updated values of 'autoload' option from 'yes' or 'no' to 'on' or 'off' respectively, across several PHP files. This change was implemented in the functions handling options and lock settings, enhancing clarity and consistency of the code.

joemcgill reviewed (commented):
Thanks @pbearne. These all look right to me. I see a few places in `wp_ajax_wp_compression_test()` that could be changed as well.

There are also a few places in `upgrade.php` where the previous values are used, but it may be best to leave those alone for now, since they were written for previous WP versions prior to this change being made.

Commit:
Update script compression settings in ajax-actions

The commit modifies the 'can_compress_scripts' options in the ajax-actions.php file. The update_site_option and update_option functions now have a third argument which controls whether the setting is on or off. This enhances flexibility and control over the script compression feature.

joemcgill commented on src/wp-admin/includes/ajax-actions.php:
```diff
@@ -190,9 +190,9 @@ function wp_ajax_wp_compression_test() {
 		// Use `update_option()` on single site to mark the option for autoloading.
 		if ( is_multisite() ) {
-			update_site_option( 'can_compress_scripts', 0 );
+			update_site_option( 'can_compress_scripts', 0, 'off' );
```
`update_site_option()` does not support a third parameter for autoload values, which is why it didn't exist here and below.

pbearne replied:
yep I just caught that error

joemcgill reviewed (changes requested):
Thanks for the update. One important change that needs to be addressed.

Commit:
Update autoload parameters in option functions

The commit adjusts the parameters for 'update_option' functions in 'featured-content.php' and 'ajax-actions.php'. It changes the 'autoload' parameter where necessary, updating the values for "featured-content" and "can_compress_scripts" options. It's done to ensure the correct use of the autoloading feature in WordPress functionality.

joemcgill reviewed (approved):
Thanks, this looks good to me now.

mukeshpanchal27 commented on src/wp-content/themes/twentyfourteen/inc/featured-content.php:
```diff
@@ -268,7 +268,7 @@ public static function delete_post_tag( $tag_id ) {
 		$settings['tag-id'] = 0;
 		$settings           = self::validate_settings( $settings );
-		update_option( 'featured-content', $settings );
+		update_option( 'featured-content', $settings, 'on' );
```
Why `on` added here?

pbearne replied:
It is the default setting so I set it to be clear as to what is value

mukeshpanchal27 replied:
If this is the case then we should update other instances in core. WDYT?

pbearne replied:
Yes I am looking for them
I feel it is good practice to set this so it clear

pbearne replied:
I have found load more 
I have not set all to 'on' but rather used my judgment and turned off some (ie just admin settings)
So this will need to be reviewed

joemcgill replied:
Let's hold off on making any changes to default themes, since those need to support older versions of WordPress. Apologies for missing this in my last review.

mukeshpanchal27 replied:
Agree with @joemcgill because of old version support we should not update bundle theme.

pbearne replied:
I have removed and moved this change to #61103

Commit:
Adjust third argument in update_option() across various scripts

This commit adds a third argument to the update_option() function calls across a large number of scripts. This argument toggles whether the option should be loaded from cache (on) or not (off). The changes made help to better manage data caching in WordPress and potentially improve performance by avoiding unnecessary database queries.

joemcgill reviewed (changes requested):
I missed the change to twentyfourteen in my last review. Can we remove that change and also not change any instances of `update_option()` that were previously using the default value? 

The scope of this ticket should just be to update places in core where a third parameter was defined, but was using the old "true|false" values.

Commit:
Adjust third argument in update_option() across various scripts

This commit adds a third argument to the update_option() function calls across a large number of scripts. This argument toggles whether the option should be loaded from cache (on) or not (off). The changes made help to better manage data caching in WordPress and potentially improve performance by avoiding unnecessary database queries.

Commit:
revert of block autoload settings

joemcgill reviewed (approved):
Thanks for updating this, @pbearne. This looks like a good update.

joemcgill commented:
Merged in https://core.trac.wordpress.org/changeset/58105.
